2007-07-20  Jakub Jelinek  <jakub@redhat.com>

	[BZ #4813]
	* login/forkpty.c (forkpty): Close master and slave fds on
	fork failure.  Patch by
	Mikulas Patocka <mikulas@artax.karlin.mff.cuni.cz>.

--- libc/login/forkpty.c	6 Jul 2001 04:55:34 -0000	1.2
+++ libc/login/forkpty.c	20 Jul 2007 07:20:07 -0000	1.3
@@ -38,6 +38,8 @@ forkpty (amaster, name, termp, winp)
   switch (pid = fork ())
     {
     case -1:
+      close (master);
+      close (slave);
       return -1;
     case 0:
       /* Child.  */
